### PR TITLE
Don't let users add non-GSMish characters to sms templates

### PR DESCRIPTION
**Pull request by leohemsted**
- [x] https://github.com/alphagov/notifications-utils/pull/116
- [x] https://github.com/alphagov/notifications-admin/pull/1128

image

If the character can be downgraded safely (like the `“”` characters or non-breaking spaces etc) then that's fine, but if it can't, like emojis or non-latin characters, then we should error, because otherwise we'll be billed inconsistent amounts by our providers

Email templates are unaffected

The downgradeable characters are still persisted to the database

### CODE REVIEW

File: tests/app/main/test_validators.py [hunk 1]
```diff
@@ -141,6 +141,21 @@ def test_for_commas_in_placeholders(
...
+def test_gsm_character_validation(client, msg):
+    OnlyGSMCharacters()(None, _gen_mock_field(msg))
+
+
+def test_non_gsm_character_validation(client):
+    with pytest.raises(ValidationError) as error:
+        OnlyGSMCharacters()(None, _gen_mock_field('∆ abc 📲 def 📵 ghi'))
+
+    assert str(error.value) == (
+        'You can’t use ‘∆’, ‘📲’ or ‘📵’ in text messages. '
```
Comment (quis) [inline tests/app/main/test_validators.py]: > You can’t use ∆, 📲 or 📵 in text messages.

I think for single characters it’s clearer without the surrounding quotation marks.

File: tests/app/main/test_validators.py [hunk 2]
```diff
@@ -141,6 +141,21 @@ def test_for_commas_in_placeholders(
...
+        'They won’t show up properly on everyone’s phones.'
```
Comment (quis) [inline tests/app/main/test_validators.py]: Will this say ‘it won’t show up properly…’ if there’s only one errant character?

Comment (leohemsted) [inline tests/app/main/test_validators.py]: damn no it won't

**Comment by quis** [comment]
Utils changes got brought in as part of https://github.com/alphagov/notifications-admin/pull/1137

Will resolve conflicts now…

**Comment by quis** [comment]
Should be updated to include https://github.com/alphagov/notifications-utils/pull/119/files

**Pull request merged**